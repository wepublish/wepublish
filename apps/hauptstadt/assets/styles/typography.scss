@import '~vuetify/src/styles/settings/variables';

/**
############
  TYPOGRAPHY
############
 */
$min-font-size: 10;
$max-font-size: 66;
@mixin font-base() {
  font-weight: 400;
  font-style: normal;
  font-variant-caps: normal;
  font-variant-east-asian: normal;
  font-variant-ligatures: normal;
  font-variant-numeric: normal;
  font-stretch: 100%;
}

@mixin title-base($font-size) {
  line-height: 1.2;
  font-size: #{$font-size}px;
  @include font-base;
}
@mixin lead-base($font-size) {
  line-height: 1.25;
  font-size: #{$font-size}px;
  @include font-base;
}
@mixin caption-base($font-size) {
  line-height: 1.3;
  font-size: #{$font-size}px;
  @include font-base;
}

/*
  TITLE
 */
@for $font-size from $min-font-size through $max-font-size {
  .title-#{$font-size} {
    @include title-base($font-size);
  }
  @each $breakpoint-name, $breakpoint-width in $grid-breakpoints {
    @media only screen and (min-width: #{$breakpoint-width}) {
      .title-#{$breakpoint-name}-#{$font-size} {
        line-height: 1.2 !important;
        font-size: #{$font-size}px !important;
        @include font-base;
      }
    }
  }
}

/*
  LEAD
 */
@for $font-size from $min-font-size through $max-font-size {
  .lead-#{$font-size} {
    @include lead-base($font-size);
  }
  @each $breakpoint-name, $breakpoint-width in $grid-breakpoints {
    @media only screen and (min-width: #{$breakpoint-width}) {
      .lead-#{$breakpoint-name}-#{$font-size} {
        line-height: 1.25 !important;
        font-size: #{$font-size}px !important;
        @include font-base;
      }
    }
  }
}

/**
  CAPTION
 */
@for $font-size from $min-font-size through $max-font-size {
  .caption-#{$font-size} {
    @include caption-base($font-size);
  }
  @each $breakpoint-name, $breakpoint-width in $grid-breakpoints {
    @media only screen and (min-width: #{$breakpoint-width}) {
      .caption-#{$breakpoint-name}-#{$font-size} {
        line-height: 1.3 !important;
        font-size: #{$font-size}px !important;
        @include font-base;
      }
    }
  }
}

/**
  FONT SIZES
 */
@for $font-size from $min-font-size through $max-font-size {
  .font-size-#{$font-size} {
    font-size: #{$font-size}px;
    line-height: 1.5;
  }

  @each $breakpoint-name, $breakpoint-width in $grid-breakpoints {
    @media only screen and (min-width: #{$breakpoint-width}) {
      .font-size-#{$breakpoint-name}-#{$font-size} {
        font-size: #{$font-size}px !important;
        line-height: 1.5;
      }
    }
  }
}


/**
  LINE HEIGHTS
 */
.line-height-1-6 {
  line-height: 1.6 !important;
}

.line-height-1-35 {
  line-height: 1.35 !important;
}

.line-height-1-25 {
  line-height: 1.25 !important;
}
.line-height-16 {
  line-height: 16px;
}

/**
  FONT FAMILY TYPES
 */

$font-families: (
  'tiempos': 'tiempos, serif',
  'tiempos-semibold': 'tiempos semibold, serif',
  'tiempos-medium': 'tiempos medium, serif',
  'tiempos-bold': 'tiempos bold, serif',
  'tiempos-bold-italic': 'tiempos bold italic, serif',
  'abc': 'ABCWhyte, sans-serif',
  'abc-light': 'ABCWhyte light, sans-serif',
  'abc-thin': 'ABCWhyte thin, sans-serif',
  'abc-bold': 'ABCWhyte bold, sans-serif',
  'abc-bold-italic': 'ABCWhyte bold italic, sans-serif',
  'abc-black': 'ABCWhyte black, sans-serif',
  'abc-super': 'ABCWhyte super, sans-serif',
  'abc-heavy': 'ABCWhyte heavy, sans-serif',
  'abc-black-italic': 'ABCWhyte black italic, sans-serif',
  'abc-super-italic': 'ABCWhyte super italic, sans-serif',
  'abc-heavy-italic': 'ABCWhyte heavy italic, sans-serif'
);

@each $class-name, $font-family in $font-families {
  .#{$class-name} {
    font-family: #{$font-family} !important;
  }

  @each $breakpoint-name, $breakpoint-width in $grid-breakpoints {
    @media (min-width: #{$breakpoint-width}) {
      .#{$class-name}-#{$breakpoint-name} {
        font-family: #{$font-family} !important;
      }
    }
  }
}

/**
  HELPERS
 */
.word-break-keep-all {
  word-break: keep-all !important;
}

/**
ABC WHITE
 */
/* regular */
@font-face {
  font-family: 'ABCWhyte';
  font-weight: normal;
  src: url('../fonts/abc-white/ABCWhyte-Regular.woff2') format('woff2');
}

/* regular italic */
@font-face {
  font-family: 'ABCWhyte';
  font-weight: normal;
  font-style: italic;
  src: url('../fonts/abc-white/ABCWhyte-RegularItalic.woff2') format('woff2');
}

/* light */
@font-face {
  font-family: 'ABCWhyte light';
  font-weight: normal;
  src: url('../fonts/abc-white/ABCWhyte-Light.woff2') format('woff2');
}

/* light italic */
@font-face {
  font-family: 'ABCWhyte light';
  font-weight: normal;
  font-style: italic;
  src: url('../fonts/abc-white/ABCWhyte-LightItalic.woff2') format('woff2');
}

/* thin */
@font-face {
  font-family: 'ABCWhyte thin';
  font-weight: 400;
  src: url('../fonts/abc-white/ABCWhyte-Thin.woff2') format('woff2');
}

/* thin italic */
@font-face {
  font-family: 'ABCWhyte thin';
  font-weight: 400;
  font-style: italic;
  src: url('../fonts/abc-white/ABCWhyte-ThinItalic.woff2') format('woff2');
}

/* bold */
@font-face {
  font-family: 'ABCWhyte bold';
  font-weight: bold;
  src: url('../fonts/abc-white/ABCWhyte-Bold.woff2') format('woff2');
}
/* bold italic */
@font-face {
  font-family: 'ABCWhyte bold italic';
  font-weight: bold;
  font-style: italic;
  src: url('../fonts/abc-white/ABCWhyte-BoldItalic.woff2') format('woff2');
}

/* bold */
@font-face {
  font-family: 'ABCWhyte black';
  font-weight: bold;
  src: url('../fonts/abc-white/ABCWhyte-Black.woff2') format('woff2');
}
/* bold italic */
@font-face {
  font-family: 'ABCWhyte black italic';
  font-weight: bold;
  font-style: italic;
  src: url('../fonts/abc-white/ABCWhyte-BlackItalic.woff2') format('woff2');
}

/* bold */
@font-face {
  font-family: 'ABCWhyte heavy';
  font-weight: bold;
  src: url('../fonts/abc-white/ABCWhyte-Heavy.woff2') format('woff2');
}
/* bold italic */
@font-face {
  font-family: 'ABCWhyte heavy italic';
  font-weight: bold;
  font-style: italic;
  src: url('../fonts/abc-white/ABCWhyte-HeavyItalic.woff2') format('woff2');
}

/* bold */
@font-face {
  font-family: 'ABCWhyte super';
  font-weight: bold;
  src: url('../fonts/abc-white/ABCWhyte-Super.woff2') format('woff2');
}
/* bold italic */
@font-face {
  font-family: 'ABCWhyte super italic';
  font-weight: bold;
  font-style: italic;
  src: url('../fonts/abc-white/ABCWhyte-SuperItalic.woff2') format('woff2');
}

/**
TIEMPOS
 */
/* regular */
@font-face {
  font-family: 'tiempos';
  font-weight: normal;
  src: url('assets/fonts/tiempos/tiempos-text-web-regular.woff2') format('woff2');
}
/* regular italic */
@font-face {
  font-family: 'tiempos';
  font-weight: normal;
  font-style: italic;
  src: url('assets/fonts/tiempos/tiempos-text-web-regular-italic.woff2') format('woff2');
}

/* medium */
@font-face {
  font-family: 'tiempos medium';
  font-weight: normal;
  src: url('assets/fonts/tiempos/tiempos-text-web-medium.woff2') format('woff2');
}
/* regular italic */
@font-face {
  font-family: 'tiempos medium';
  font-weight: normal;
  font-style: italic;
  src: url('assets/fonts/tiempos/tiempos-text-web-medium-italic.woff2') format('woff2');
}

/* semi-bold */
@font-face {
  font-family: 'tiempos semibold';
  src: url('assets/fonts/tiempos/tiempos-text-web-semibold.woff2') format('woff2');
}
/* semi-bold italic */
@font-face {
  font-family: 'tiempos semibold';
  font-style: italic;
  src: url('assets/fonts/tiempos/tiempos-text-web-semibold-italic.woff2') format('woff2');
}
/* bold */
@font-face {
  font-family: 'tiempos bold';
  font-weight: normal;
  src: url('assets/fonts/tiempos/tiempos-text-web-bold.woff2') format('woff2');
}
/* bold italic */
@font-face {
  font-family: 'tiempos bold italic';
  font-weight: normal;
  src: url('assets/fonts/tiempos/tiempos-text-web-bold-italic.woff2') format('woff2');
}
