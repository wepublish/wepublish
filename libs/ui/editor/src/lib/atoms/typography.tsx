import * as CSS from 'csstype'
import {ElementType, forwardRef, ReactNode} from 'react'

export type TypographyVariant =
  | 'title'
  | 'h1'
  | 'h2'
  | 'h3'
  | 'body1'
  | 'body2'
  | 'subtitle1'
  | 'subtitle2'

export type TypographyTextAlign = 'left' | 'center' | 'right'
export type TypographyDisplay = 'block' | 'inline'
export type TypographySpacing = 'small' | 'large'

export interface TypographyProps extends React.ComponentPropsWithRef<'div'> {
  variant?: TypographyVariant
  color?: string
  align?: TypographyTextAlign
  display?: TypographyDisplay
  spacing?: TypographySpacing
  ellipsize?: boolean
  element?: ElementType<{className?: string}>
  children?: ReactNode
}

export const Typography = forwardRef<HTMLElement, TypographyProps>(function Typography(
  {
    variant = 'body1',
    color,
    align,
    display,
    spacing,
    ellipsize,
    element = elementForTypographyVariant(variant),
    children,
    ...props
  },
  ref
) {
  const Element = element as any

  return (
    <Element
      ref={ref}
      style={{
        display,
        textAlign: align,
        color: color || undefined,
        fill: color || undefined,
        whiteSpace: ellipsize ? 'nowrap' : undefined,
        textOverflow: ellipsize ? 'ellipsis' : undefined,
        overflow: ellipsize ? 'hidden' : undefined,
        marginTop: 0,
        marginBottom: spacing ? marginForTypographySpacing(spacing) : 0,
        ...stylesForTypographyVariant(variant)
      }}
      {...props}>
      {children}
    </Element>
  )
})

export function elementForTypographyVariant(variant: TypographyVariant) {
  switch (variant) {
    case 'title':
      return 'h1'

    case 'h1':
      return 'h1'

    case 'h2':
      return 'h2'

    case 'h3':
      return 'h3'

    default:
      return 'p'
  }
}

export function marginForTypographySpacing(spacing: TypographySpacing): string {
  switch (spacing) {
    case 'small':
      return '0.4em'

    case 'large':
      return '0.8em'
  }
}

export interface TypographyVariantCSS {
  readonly fontSize?: CSS.Property.FontSize
  readonly fontWeight?: CSS.Property.FontWeight
  readonly fontStyle?: CSS.Property.FontStyle
}

export function stylesForTypographyVariant(style: TypographyVariant): TypographyVariantCSS {
  switch (style) {
    case 'title':
      return {
        fontSize: '40px',
        fontWeight: 'bold'
      }

    case 'h1':
      return {
        fontSize: '28px',
        fontWeight: 'bold'
      }

    case 'h2':
      return {
        fontSize: '24px',
        fontWeight: 'bold'
      }

    case 'h3':
      return {
        fontSize: '20px',
        fontWeight: 'bold'
      }

    case 'body1':
    case 'body2':
      return {
        fontSize: '16px',
        fontWeight: style === 'body2' ? 'bold' : undefined
      }

    case 'subtitle1':
    case 'subtitle2':
      return {
        fontSize: '12px',
        fontStyle: style === 'subtitle2' ? 'italic' : undefined
      }
  }
}
