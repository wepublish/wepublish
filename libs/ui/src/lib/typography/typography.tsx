import {Typography} from '@mui/material'
import {AnchorHTMLAttributes, forwardRef, HTMLAttributes} from 'react'
import {Link as MuiLink, LinkTypeMap} from '@mui/material'

export type HeadingProps = HTMLAttributes<HTMLHeadingElement> & {component?: React.ElementType}

export const H1 = forwardRef(({children, ...props}: HeadingProps, ref) => {
  return (
    <Typography {...props} ref={ref as any} variant="h1">
      {children}
    </Typography>
  )
})

export const H2 = forwardRef(({children, ...props}: HeadingProps, ref) => {
  return (
    <Typography {...props} ref={ref as any} variant="h2">
      {children}
    </Typography>
  )
})

export const H3 = forwardRef(({children, ...props}: HeadingProps, ref) => {
  return (
    <Typography {...props} ref={ref as any} variant="h3">
      {children}
    </Typography>
  )
})

export const H4 = forwardRef(({children, ...props}: HeadingProps, ref) => {
  return (
    <Typography {...props} ref={ref as any} variant="h4">
      {children}
    </Typography>
  )
})

export const H5 = forwardRef(({children, ...props}: HeadingProps, ref) => {
  return (
    <Typography {...props} ref={ref as any} variant="h5">
      {children}
    </Typography>
  )
})

export const H6 = forwardRef(({children, ...props}: HeadingProps, ref) => {
  return (
    <Typography {...props} ref={ref as any} variant="h6">
      {children}
    </Typography>
  )
})

export type ParagraphProps = HTMLAttributes<HTMLParagraphElement> & {component?: React.ElementType}

export const Paragraph = forwardRef(({children, ...props}: ParagraphProps, ref) => {
  return (
    <Typography {...props} ref={ref as any} variant="body1">
      {children}
    </Typography>
  )
})

export type LinkProps = AnchorHTMLAttributes<HTMLAnchorElement> & {
  component?: React.ElementType
  underline?: LinkTypeMap['props']['underline']
  color?: LinkTypeMap['props']['color']
}

export const Link = forwardRef(({children, underline, color, ...props}: LinkProps, ref) => {
  return (
    <MuiLink {...props} ref={ref as any} variant="body1" color={color} underline={underline}>
      {children}
    </MuiLink>
  )
})
