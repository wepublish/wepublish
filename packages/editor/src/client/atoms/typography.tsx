import React, {CSSProperties, ElementType, forwardRef, ReactNode} from 'react'

export type TypographyVariant =
  | 'title'
  | 'h1'
  | 'h2'
  | 'h3'
  | 'body1'
  | 'body2'
  | 'subtitle1'
  | 'subtitle2'

export type TypographyTextAlign = 'left' | 'center' | 'right'
export type TypographyDisplay = 'block' | 'inline'
export type TypographySpacing = 'small' | 'large'

export interface TypographyProps {
  variant?: TypographyVariant
  color?: string
  align?: TypographyTextAlign
  display?: TypographyDisplay
  spacing?: TypographySpacing
  ellipsize?: boolean
  element?: ElementType<{className?: string}>
  children?: ReactNode
}

export const Typography = forwardRef<HTMLElement, TypographyProps>(function Typography(
  {
    variant = 'body1',
    color,
    align,
    display,
    spacing,
    ellipsize,
    element = elementForTypographyVariant(variant),
    children,
    ...props
  },
  ref
) {
  const Element = element as any

  return (
    <Element
      ref={ref}
      style={{
        display,
        textAlign: align,
        color: color || undefined,
        fill: color || undefined,
        whiteSpace: ellipsize ? 'nowrap' : undefined,
        textOverflow: ellipsize ? 'ellipsis' : undefined,
        overflow: ellipsize ? 'hidden' : undefined,
        marginTop: 0,
        marginBottom: spacing ? marginForTypographySpacing(spacing) : 0,
        ...stylesForTypographyVariant(variant)
      }}
      {...props}>
      {children}
    </Element>
  )
})

export function elementForTypographyVariant(variant: TypographyVariant) {
  switch (variant) {
    case 'title':
      return 'h1'

    case 'h1':
      return 'h1'

    case 'h2':
      return 'h2'

    case 'h3':
      return 'h3'

    default:
      return 'p'
  }
}

export function marginForTypographySpacing(spacing: TypographySpacing): string {
  switch (spacing) {
    case 'small':
      return '0.4em'

    case 'large':
      return '0.8em'
  }
}

export function stylesForTypographyVariant(style: TypographyVariant): CSSProperties {
  switch (style) {
    case 'title':
      return {
        fontSize: 40,
        fontWeight: 'bold'
      }

    case 'h1':
      return {
        fontSize: 28,
        fontWeight: 'bold'
      }

    case 'h2':
      return {
        fontSize: 24,
        fontWeight: 'bold'
      }

    case 'h3':
      return {
        fontSize: 20,
        fontWeight: 'bold'
      }

    case 'body1':
    case 'body2':
      return {
        fontSize: 16,
        fontWeight: style === 'body2' ? 'bold' : undefined
      }

    case 'subtitle1':
    case 'subtitle2':
      return {
        fontSize: 12,
        fontStyle: style === 'subtitle2' ? 'italic' : undefined
      }
  }
}
