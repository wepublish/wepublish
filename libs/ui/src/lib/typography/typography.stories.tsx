import {ComponentMeta} from '@storybook/react'
import {Button, ButtonProps} from './typography'

export default {
  component: Button,
  title: 'Button'
} as ComponentMeta<typeof Button>

const Template = (args: ButtonProps) => <Button {...args}>Click me</Button>
export const Default = Template.bind({})
