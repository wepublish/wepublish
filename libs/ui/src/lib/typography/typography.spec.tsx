import {render} from '@testing-library/react'

import {H1, H2, H3, Paragraph} from './typography'

describe('H1', () => {
  it('should render successfully', () => {
    const {baseElement} = render(<H1>Foobar</H1>)
    expect(baseElement).toMatchSnapshot()
  })
})

describe('H2', () => {
  it('should render successfully', () => {
    const {baseElement} = render(<H2>Foobar</H2>)
    expect(baseElement).toMatchSnapshot()
  })
})

describe('H3', () => {
  it('should render successfully', () => {
    const {baseElement} = render(<H3>Foobar</H3>)
    expect(baseElement).toMatchSnapshot()
  })
})

describe('Paragraph', () => {
  it('should render successfully', () => {
    const {baseElement} = render(<Paragraph>Foobar</Paragraph>)
    expect(baseElement).toMatchSnapshot()
  })
})
