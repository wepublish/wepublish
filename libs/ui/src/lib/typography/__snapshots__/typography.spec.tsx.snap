// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 should render successfully 1`] = `
<body>
  <div>
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Foobar
    </h1>
  </div>
</body>
`;

exports[`H2 should render successfully 1`] = `
<body>
  <div>
    <h2
      class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
    >
      Foobar
    </h2>
  </div>
</body>
`;

exports[`H3 should render successfully 1`] = `
<body>
  <div>
    <h3
      class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    >
      Foobar
    </h3>
  </div>
</body>
`;

exports[`Paragraph should render successfully 1`] = `
<body>
  <div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Foobar
    </p>
  </div>
</body>
`;
